### PR TITLE
[CELEBORN-1041] Improve the implementation for get the PartitionIdPassthrough class

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
Currently, the code of get the contractor of `PartitionIdPassthrough` is very redundant.
We should improve the implementation.

### Why are the changes needed?
Improve the implementation for get the `PartitionIdPassthrough` class

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
New test cases.

### CODE REVIEW

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks! Merging to main/0.3

File: client-spark/spark-3/src/test/java/org/apache/spark/shuffle/celeborn/CelebornShuffleWriterSuiteBase.java [hunk 1]
```diff
@@ -217,13 +218,14 @@ private void check(
...
+      PartitionIdPassthroughClazz = "org.apache.spark.sql.execution.PartitionIdPassthrough";
+    }
     DynConstructors.Ctor<Partitioner> partitionIdPassthroughCtor =
-        DynConstructors.builder()
-            // for Spark 3.3 and previous
-            .impl("org.apache.spark.sql.execution.PartitionIdPassthrough", int.class)
-            // for Spark 3.4
-            .impl("org.apache.spark.PartitionIdPassthrough", int.class)
-            .build();
+        DynConstructors.builder().impl(PartitionIdPassthroughClazz, int.class).build();
```
Comment (cfmcgrady) [inline client-spark/spark-3/src/test/java/org/apache/spark/shuffle/celeborn/CelebornShuffleWriterSuiteBase.java:228]: Actually, I think the original implementation has more readability

Comment (beliefer) [reply client-spark/spark-3/src/test/java/org/apache/spark/shuffle/celeborn/CelebornShuffleWriterSuiteBase.java:228]: I think the both should have the same readability.
But the original implementation called 'impl' twice was confusing.

**Comment by beliefer** [comment]
@waitinfuture @cfmcgrady Thank you!

**Pull request closed without merging**